CliClient/tests/services_SearchFilter.js

```diff
@@ -0,0 +1,195 @@
+
+		await engine.syncTables();
+		rows = await engine.search('title: abcd');
+
+		expect(rows.length).toBe(1);
+		expect(rows[0].id).toBe(n1.id);
+	}));
+
+	it('should return note matching body', asyncTest(async () => {
+		let rows;
+		const n1 = await Note.save({ title: 'abcd', body: 'body1' });
+		const n2 = await Note.save({ title: 'efgh', body: 'body2' });
+
+		await engine.syncTables();
+		rows = await engine.search('body: body1');
+
+		expect(rows.length).toBe(1);
+		expect(rows[0].id).toBe(n1.id);
+	}));
+
+	it('should return note matching title containing multiple words', asyncTest(async () => {
+		let rows;
+		const n1 = await Note.save({ title: 'abcd xyz', body: 'body1' });
+		const n2 = await Note.save({ title: 'efgh ijk', body: 'body2' });
+
+		await engine.syncTables();
+		rows = await engine.search('title: "abcd xyz"');
+
+		expect(rows.length).toBe(1);
+		expect(rows[0].id).toBe(n1.id);
+	}));
+
+	it('should return note matching body containing multiple words', asyncTest(async () => {
+		let rows;
+		const n1 = await Note.save({ title: 'abcd', body: 'ho ho ho' });
+		const n2 = await Note.save({ title: 'efgh', body: 'foo bar' });
+
+		await engine.syncTables();
+		rows = await engine.search('body: "foo bar"');
+
+		expect(rows.length).toBe(1);
+		expect(rows[0].id).toBe(n2.id);
+	}));
+
+	it('should return note matching title AND body', asyncTest(async () => {
+		let rows;
+		const n1 = await Note.save({ title: 'abcd', body: 'ho ho ho' });
+		const n2 = await Note.save({ title: 'efgh', body: 'foo bar' });
+
+		await engine.syncTables();
+		rows = await engine.search('title: efgh body: "foo bar"');
+		expect(rows.length).toBe(1);
+		expect(rows[0].id).toBe(n2.id);
+
+		rows = await engine.search('title: abcd body: "foo bar"');
+		expect(rows.length).toBe(0);
+	}));
+
+	it('should return note matching title OR body', asyncTest(async () => {
+		let rows;
+		const n1 = await Note.save({ title: 'abcd', body: 'ho ho ho' });
+		const n2 = await Note.save({ title: 'efgh', body: 'foo bar' });
+
+		await engine.syncTables();
+		rows = await engine.search('title: abcd OR body: "foo bar"');
+		expect(rows.length).toBe(2);
+		expect(rows.map(r=>r.id)).toContain(n1.id);
+		expect(rows.map(r=>r.id)).toContain(n2.id);
+
+		rows = await engine.search('title: wxyz OR body: "blah blah"');
+		expect(rows.length).toBe(0);
+	}));
+
+	it('should return notes matching text', asyncTest(async () => {
+		let rows;
+		const n1 = await Note.save({ title: 'foo beef', body: 'dead bar' });
+		const n2 = await Note.save({ title: 'bar efgh', body: 'foo dog' });
+		await engine.syncTables();
+
+		// Interpretation: Match with notes containing foo in title/body and bar in title/body
+		// Note: This is NOT saying to match notes containing foo bar in title/body
+		rows = await engine.search('foo bar');
+		expect(rows.length).toBe(2);
+		expect(rows.map(r=>r.id)).toContain(n1.id);
+		expect(rows.map(r=>r.id)).toContain(n2.id);
+
```

Comment:
It's good you use `toContain` as indeed the order may change depending on various factors which are not relevant to this test. 👍